interp/interp.go

```diff
@@ -57,6 +58,7 @@ type frame struct {
 	recovered interface{}       // to handle panic recover
 	runid     uint64            // for cancellation
 	done      chan struct{}     // for cancellation of channel operations
+	mutex     sync.Mutex
```

Comment:
This should be placed above the fields that is protects, with an annotation listing those fields it is protecting.

Also, I'd suggest this be a `RWMutex` since that will save most contended cases.

Reply:
Actually this should have been a comment on `Interpreter`.